packages/core/src/physics/Collider.ts

```diff
@@ -0,0 +1,91 @@
+import { Component } from "../Component";
+import { ignoreClone } from "../clone/CloneManager";
+import { ICollider } from "@oasis-engine/design";
+import { Vector3 } from "@oasis-engine/math";
+import { ColliderShape } from "./shape/ColliderShape";
+import { DisorderedArray } from "../DisorderedArray";
+import { UpdateFlag } from "../UpdateFlag";
+import { Entity } from "../Entity";
+
+/**
+ * Abstract class for collider shapes.
+ */
+export abstract class Collider extends Component {
+  protected static _tempScale: Vector3 = new Vector3();
+  /** @internal */
+  @ignoreClone
+  _index: number = -1;
+  /** @internal */
+  _nativeCollider: ICollider;
+
+  private _shapes: DisorderedArray<ColliderShape> = new DisorderedArray();
```

Comment:
`DisorderedArray` is uesd for high-performance with large number of elements and no requirement on the order of elements.  But shapes is a public API，must ensure the order is correct.

Reply:
done